src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -440,7 +440,25 @@ public Forum getForumByName(String name) throws EntityNotFoundException{
    * @throws EntityNotFoundException
    */
   public List<Article> getArticlesOfForum(Forum forum) throws EntityNotFoundException {
-    List<Entity> result= queryByList(forum.getArticleIds(),"Article");
+      //Initialize empty arraylist
+      List<Entity> result = new ArrayList<>();
+
+      try {
+          //Retrieve list of article entities
+          result = queryByList(forum.getArticleIds(), "Article");
+      }
+      //An entity was not found
+      catch (EntityNotFoundException e) {
+          //Remove from the forum entity stored in datastore
+          removeFieldForum(forum.getTitle(), "articleIds", e.getKey().getName(), true);
```

Comment:
Is this a hack? Why is there "removing" happening in a get method?

Reply:
In case if there was an exception where the article was not found but is nevertheless stored on a forum it would remove the article from the forum and update it on the datastore.